packages/core-base/src/runtime.ts

```diff
@@ -21,8 +22,38 @@ type StringConvertable<T> = ExtractToStringKey<T> extends never
 	  ? T
 	  : unknown
 
+/**
+ *
+ * The interface used for narrowing types using generated types.
+ *
+ * @remarks
+ *
+ * The type generated by 3rd party (e.g. nuxt/i18n)
+ *
+ * @example
+ * ```ts
+ * // generated-intlify-core.d.ts (`.d.ts` file at your app)
+ *
+ * declare module '@intlify/core' {
+ *   interface IntlifyGeneratedTypeConfig {
+ *     locale: "en" | "ja"
+ *   }
+ * }
+ * ```
+ */
+export interface IntlifyGeneratedTypeConfig {}
```

Comment:
We don't need the `Intlify` prefix, because we currently do not have that prefix in the `core` and `core-base` type definitions. :)